src/applications/vaos/tests/services/appointment/index.v2.unit.spec.js

```diff
@@ -730,4 +733,407 @@ describe('VAOS Appointment service', () => {
+              ...data,
+            }),
+          ],
+        },
+      );
+
+      const [v0Result, v2Result] = await Promise.all([
+        getAppointmentRequests({
+          startDate,
+          endDate,
+        }),
+        getAppointmentRequests({
+          startDate,
+          endDate,
+          useV2: true,
+        }),
+      ]);
+
+      // These are always different
+      delete v0Result[0].vaos.apiData;
+      delete v2Result[0].vaos.apiData;
+
+      // Then expect these differences
+      // differences format is http://jsonpatch.com/
+      const differences = diff(v2Result[0], v0Result[0]);
+      expect(differences).to.have.deep.members(
+        [
+          { op: 'remove', path: ['description'] },
+          { op: 'remove', path: ['practitioners'] },
+          {
+            op: 'remove',
+            path: ['vaos', 'isPastAppointment'],
+          },
+          {
+            op: 'remove',
+            path: ['vaos', 'isPhoneAppointment'],
+          },
+          {
+            op: 'remove',
+            path: ['vaos', 'isCOVIDVaccine'],
+          },
+          {
+            op: 'remove',
+            path: ['vaos', 'timeZone'],
+          },
+          {
+            op: 'replace',
+            path: ['created'],
+            value: moment().format('YYYY-MM-DD'),
+          },
+          {
+            op: 'replace',
+            path: ['location', 'clinicId'],
+            value: null,
+          },
+          { op: 'add', path: ['cancelationReason'], value: null },
+        ],
+        'Transformers for v0 and v2 appointment request data are out of sync',
+      );
+    });
+
+    it('should return matching v0 and v2 data for a CC appointment request', async () => {
+      // Given CC appointment request data
+      const data = {
+        id: '1234',
+        email: 'test@va.gov',
+        phone: '2125551212',
+        kind: 'cc',
+        clinicFriendlyName: 'Friendly clinic name',
+        requestedPeriods: [
+          {
+            start: `${moment().format('YYYY-MM-DD')}T00:00:00.000`,
+            end: `${moment().format('YYYY-MM-DD')}T11:59:59.999`,
+          },
+        ],
+        serviceType: 'primaryCare',
+        status: 'proposed',
+        typeOfCareId: 'CCPRMYRTNE',
+        visitType: 'Office Visit',
+      };
+      const startDate = moment()
+        .subtract(30, 'days')
+        .format();
+      const endDate = moment()
+        .add(30, 'days')
+        .format();
+
+      // When we request v2 appointment requests
+      setFetchJSONResponse(
+        global.fetch.withArgs(
+          sinon.match(
+            `/vaos/v2/appointments?start=${startDate}&end=${endDate}&statuses[]=proposed&statuses[]=cancelled`,
+          ),
+        ),
+        {
+          data: [
+            createMockAppointmentByVersion({
+              version: 2,
+              ...data,
+            }),
+          ],
+        },
+      );
+
+      // And When we request v0 appointment requests
```

Comment:
Generally we should only have one When comment, since there should be [one thing](https://automationpanda.com/2018/02/03/are-gherkin-scenarios-with-multiple-when-then-pairs-okay/) that we're testing the result of. In these tests I think that one thing is the comparison between the two calls. So maybe:

```
Given a CC request
  And the developer fetched that request through both the v2 and v0 APIs
When they compare the two results
Then the results have the following differences
```

That said, these are more standard unit tests that are making api calls, so GWT is less useful here. It's still not a bad thing, but it's not as helpful as with UI tests.